File: v16.1/QueryPatients.cs
using System;
using System.Data.SqlClient;
using System.IO;

class QueryPatients
{
    static void Main(string[] args)
    {
        string configPath = args.Length > 0 ? args[0] : Path.Combine(Environment.CurrentDirectory, "..", "_config.json");
        string configText = File.ReadAllText(configPath);

        string server = ExtractJsonValue(configText, "db_server");
        string uid = ExtractJsonValue(configText, "db_user");
        string pw = ExtractJsonValue(configText, "db_pw");

        string connStr = string.Format(
            "Data Source={0};Initial Catalog=VARIAN;Integrated Security=False;User Id={1};Password={2};Connection Timeout=10",
            server, uid, pw);

        Console.WriteLine("Connecting to {0} as {1}...", server, uid);

        using (var con = new SqlConnection(connStr))
        {
            con.Open();
            Console.WriteLine("Connected.\n");

            string sql = "SELECT TOP 10 PatientSer, PatientId, LastName, FirstName FROM Patient ORDER BY PatientSer";
            using (var cmd = new SqlCommand(sql, con))
            using (var dr = cmd.ExecuteReader())
            {
                Console.WriteLine("{0,-12} {1,-15} {2,-20} {3,-20}", "PatientSer", "PatientId", "LastName", "FirstName");
                Console.WriteLine(new string('-', 70));
                while (dr.Read())
                {
                    Console.WriteLine("{0,-12} {1,-15} {2,-20} {3,-20}",
                        dr["PatientSer"],
                        dr.IsDBNull(1) ? "" : dr["PatientId"],
                        dr.IsDBNull(2) ? "" : dr["LastName"],
                        dr.IsDBNull(3) ? "" : dr["FirstName"]);
                }
            }
        }
    }

    static string ExtractJsonValue(string json, string key)
    {
        int idx = json.IndexOf("\"" + key + "\"");
        if (idx < 0) return "";
        idx = json.IndexOf(":", idx);
        if (idx < 0) return "";
        idx = json.IndexOf("\"", idx);
        if (idx < 0) return "";
        int end = json.IndexOf("\"", idx + 1);
        return json.Substring(idx + 1, end - idx - 1);
    }
}

File: v16.1/query_patients.py
"""Simple test query: list top 10 patients."""

import sys

from db import get_config, connect


def main():
    config = get_config()
    print(f"Connecting to {config['server']} as {config['user']}...")
    conn = connect(config)
    print("Connected.\n")

    cursor = conn.cursor()
    cursor.execute("SELECT TOP 10 PatientSer, PatientId, LastName, FirstName FROM Patient ORDER BY PatientSer")

    print(f"{'PatientSer':<12} {'PatientId':<15} {'LastName':<20} {'FirstName':<20}")
    print("-" * 70)
    for row in cursor.fetchall():
        ser, pid, last, first = row
        print(f"{ser!s:<12} {(pid or ''):<15} {(last or ''):<20} {(first or ''):<20}")

    conn.close()


if __name__ == "__main__":
    main()
